src/Azure.Functions.Cli/Common/StaticSettings.cs

```diff
@@ -1,15 +1,24 @@
 ﻿using Azure.Functions.Cli.Helpers;
 using System;
+using System.IO;
+using System.Reflection;
 
 namespace Azure.Functions.Cli.Common
 {
     public static class StaticSettings
     {
+        private static bool? _isTelemetryEnabledCache = null;
+
         public static bool IsDebug => Environment.GetEnvironmentVariable(Constants.CliDebug) == "1";
 
-        // Turning this off for now
-        // TODO: Enable this once it's ready to be released.
-        public static bool IsTelemetryEnabled => false && (string.IsNullOrEmpty(Environment.GetEnvironmentVariable(Constants.TelemetryOptOutVariable))
-                && Constants.TelemetryInstrumentationKey != "00000000-0000-0000-0000-000000000000");
+        public static bool IsTelemetryEnabled
+        {
+            get
+            {
+                bool telemetryVal = _isTelemetryEnabledCache ?? TelemetryHelpers.CheckIfTelemetryEnabled();
+                _isTelemetryEnabledCache = telemetryVal;
+                return telemetryVal;
```

Comment:
This pattern in .NET is encapsulated in [`Lazy<T>`](https://docs.microsoft.com/en-us/dotnet/framework/performance/lazy-initialization). This is not relevant for the cli since it's mostly single threaded, but this isn't thread safe either. You'd need to use a lock, but `Lazy<T>` gives you for free.

Reply:
Ooooh! Didn't know Lazy was thread safe. I guess that makes sense. Cool! I will update it. Figured if we are doing this check multiple times in one execution, we probably don't need to do this file check everytime.
I will update. Thanks!